client/src/components/History/CurrentHistory/HistoryPanel.vue

```diff
@@ -416,7 +411,7 @@ export default {
             if (data) {
                 const dataSource = data.history_content_type === "dataset" ? "hda" : "hdca";
                 if (data.history_id != this.historyId) {
-                    copyDataset(data.id, this.historyId, data.history_content_type, dataSource)
+                    await copyDataset(data.id, this.historyId, data.history_content_type, dataSource)
                         .then(() => {
                             if (data.history_content_type === "dataset") {
```

Comment:
If you use `await` you can replace the `then/catch` with a try/catch and increase a bit readability while reducing nesting.